src/Options.jl

```diff
@@ -353,6 +356,22 @@ function Options(;
     binary_operators = map(binopmap, binary_operators)
     unary_operators = map(unaopmap, unary_operators)
 
+    diff_binary_operators = Any[]
+    diff_unary_operators = Any[]
+
+    for op in binary_operators
+        diff_op(x,y) = gradient(op,x,y)
+        push!(diff_binary_operators, diff_op)
+    end
+
+    for op in unary_operators
+       diff_op(x) = sum(gradient(op,x))
```

Comment:
Why is this one summed?

Reply:
It was a debug thing that I forgot to change it back. Tho I think it doesn't matter whether it is gradient(op,x)[1] or sum(gradient(op,x)) since there is only one component in the result